internals-js/src/error.ts

```diff
@@ -306,6 +306,26 @@ const SATISFIABILITY_ERROR = makeCodeDefinition(
   'Subgraphs can be merged, but the resulting supergraph API would have queries that cannot be satisfied by those subgraphs.',
 );
 
+const OVERRIDE_FROM_SELF_ERROR = makeCodeDefinition(
+  'OVERRIDE_FROM_SELF_ERROR',
+  'Field with `@override` directive has "from" location that references its own subgraph.',
+);
+
+const OVERRIDE_SOURCE_HAS_OVERRIDE = makeCodeDefinition(
+  'OVERRIDE_SOURCE_HAS_OVERRIDE',
+  'Field which is overridden to another subgraph is also marked @override.',
+);
+
+const OVERRIDE_ON_BOTH_FIELD_AND_TYPE = makeCodeDefinition(
+  'OVERRIDE_ON_BOTH_FIELD_AND_TYPE',
+  'Field and type are both marked with @override directive. This is not currently supported.',
+);
+
+const OVERRIDE_COLLISION_WITH_ANOTHER_DIRECTIVE = makeCodeDefinition(
+  'OVERRIDE_COLLISION_WITH_ANOTHER_DIRECTIVE',
+  'Override cannot be used in conjunction with specific other directives.',
```

Comment:
Could be nice to list the directives that have incompatibility, instead of "specific other". I'd even go as far as something like `The @override directive cannot be used on external fields, nor to override fields with either @external, @provides or @requires`. My reasoning is that it's a tad annoying when you read doc and things are vague and you need to manually experiment to figure out what is allowed/not allowed.

Reply:
good shout.